### ISSUE TITLE
 Change license from AGPL3 to LGPL3 

### ISSUE THREAD

**Issue by magicus** [Wynntils/Wynntils#1970]
## TL;DR

If you have been pinged here, you have contributed code to Artemis. If you agree with changing license to LGPL3 for your contributions to Artemis, **please type "I agree" in a comment below**. (If you think you recently did this: we have changed the new license from GPL to LGPL, so we need a new set of agreements.)

## Background

Due to historic reasons, Artemis was created with the same license as Legacy, the [GNU Affero General Public License v3 (AGPL3)](https://www.gnu.org/licenses/agpl-3.0.en.html). This is a slightly modified version of GPL3, which prohibits public use on a server without releasing the source code.

Furthermore, the legal status of of whether modding Minecraft counts as "linking with a non-free program", and are thus prohibited under GPL, is unclear at best. The [perception in the community](https://blog.modrinth.com/p/licensing-guide) seems to be that it is not allowed.

LGPL was designed for this particular purpose. It is [GPL with additions](https://en.wikipedia.org/wiki/GNU_Lesser_General_Public_License#Differences_from_the_GPL) that explicitly make it clear that it is allowed to link with non-free programs. 

I propose that we change the Artemis license from AGPL3 to the [GNU Lesser General Public License v3 (LGPL3)](https://www.gnu.org/licenses/lgpl-3.0.en.html).

Since the code currently present in the Artemis repo was contributed under the AGPL3 license, all contributors must agree to a change. If an author does not agree, or does not reply, we might be forced to remove those commits that said author have provided.

Note that this change do not apply to other Wynntils software, such as Hades and Athena that are running on the Wynntils servers. They will keep using AGPL.

## Process

If you have been pinged in this message, please type "I agree" (or, "I do not agree") in a comment below. An Artemis maintainer will then update the list below based on the responses.

If and when all authors have agreed, we will re-license the project. If a few individual authors do not agree or does not reply, the Artemis development team might decide to remove those contributions and still go ahead with the re-license, keeping only code from those who agree.

## Contributors

Here is a complete list of all contributors to the Artemis repo, [as provided by Github](https://github.com/Wynntils/Artemis/graphs/contributors). 

All contributors has accepted.

**Legend:** ❓ = has not replied,     ✅ = has accepted,     ❌ = has not accepted

| User  | Commits  | Approval  |
|---|---|---|
| @kristofbolyai | [875 commits](https://github.com/Wynntils/Artemis/commits?author=kristofbolyai) | ✅ |
| @magicus | [639 commits](https://github.com/Wynntils/Artemis/commits?author=magicus) | ✅ |
| @Incompleteusern | [169 commits](https://github.com/Wynntils/Artemis/commits?author=Incompleteusern) | ✅ |
| @DonkeyBlaster | [95 commits](https://github.com/Wynntils/Artemis/commits?author=DonkeyBlaster) | ✅ |
| @P0keDev | [69 commits](https://github.com/Wynntils/Artemis/commits?author=P0keDev) | ✅ |
| @DevScyu | [36 commits](https://github.com/Wynntils/Artemis/commits?author=DevScyu) | ✅ |
| @san7890 | [27 commits](https://github.com/Wynntils/Artemis/commits?author=san7890) | ✅ |
| @Rafii2198 | [20 commits](https://github.com/Wynntils/Artemis/commits?author=Rafii2198) | ✅ |
| @Mcrtin | [20 commits](https://github.com/Wynntils/Artemis/commits?author=Mcrtin) | ✅ |
| @HighCrit | [16 commits](https://github.com/Wynntils/Artemis/commits?author=HighCrit) | ✅ |
| @JamieCallan117 | [15 commits](https://github.com/Wynntils/Artemis/commits?author=JamieCallan117) | ✅ |
| @HalcyonSuoh | [9 commits](https://github.com/Wynntils/Artemis/commits?author=HalcyonSuoh) | ✅ |
| @TomaSajt | [6 commits](https://github.com/Wynntils/Artemis/commits?author=TomaSajt) | ✅ |
| @byBackfish | [6 commits](https://github.com/Wynntils/Artemis/commits?author=byBackfish) | ✅ |
| @jh-devv | [5 commits](https://github.com/Wynntils/Artemis/commits?author=jh-devv) | ✅ |
| @re-vc | [4 commits](https://github.com/Wynntils/Artemis/commits?author=re-vc) | ✅ |
| @joshieman06 | [4 commits](https://github.com/Wynntils/Artemis/commits?author=joshieman06) | ✅ |
| @DalwynWasTaken | [4 commits](https://github.com/Wynntils/Artemis/commits?author=DalwynWasTaken) | ✅ |
| @mbnkwm | [3 commits](https://github.com/Wynntils/Artemis/commits?author=mbnkwm) | ✅ |
| @ShadowShift | [3 commits](https://github.com/Wynntils/Artemis/commits?author=ShadowShift) | ✅ |
| @ekgame | [3 commits](https://github.com/Wynntils/Artemis/commits?author=ekgame) | ✅ |
| @coehlrich | [3 commits](https://github.com/Wynntils/Artemis/commits?author=coehlrich) | ✅ |
| @McPlayHD | [2 commits](https://github.com/Wynntils/Artemis/commits?author=McPlayHD) | ✅ |
| @CringeAura | [2 commits](https://github.com/Wynntils/Artemis/commits?author=CringeAura) | ✅ |
| @Zatzou | [2 commits](https://github.com/Wynntils/Artemis/commits?author=Zatzou) | ✅ |
| @Jimmy-sheep | [2 commits](https://github.com/Wynntils/Artemis/commits?author=Jimmy-sheep) | ✅ |
| @ENORMOUZ | [2 commits](https://github.com/Wynntils/Artemis/commits?author=ENORMOUZ) | ✅ |
| @bricefrisco | [2 commits](https://github.com/Wynntils/Artemis/commits?author=bricefrisco) | ✅ |
| @YouSoPugly | [2 commits](https://github.com/Wynntils/Artemis/commits?author=YouSoPugly) | ✅ |
| @xCykrix | [1 commit](https://github.com/Wynntils/Artemis/commits?author=xCykrix) | ✅ |
| @atiedebee | [1 commit](https://github.com/Wynntils/Artemis/commits?author=atiedebee) | ✅ |
| @TUsama | [1 commit](https://github.com/Wynntils/Artemis/commits?author=TUsama) | ✅ |
| @Ldude162 | [1 commit](https://github.com/Wynntils/Artemis/commits?author=Ldude162) | ✅ |
| @cmkohnen | [1 commit](https://github.com/Wynntils/Artemis/commits?author=cmkohnen) | ✅ |
| @obj-obj | [1 commit](https://github.com/Wynntils/Artemis/commits?author=obj-obj) | ✅ |
| @iwei20 | [1 commit](https://github.com/Wynntils/Artemis/commits?author=iwei20) | ✅ |
| @DevChromium | [1 commit](https://github.com/Wynntils/Artemis/commits?author=DevChromium) | ✅ |
| @KaspianDev | [1 commit](https://github.com/Wynntils/Artemis/commits?author=KaspianDev) | ✅ |
| @geekazodium | [1 commit](https://github.com/Wynntils/Artemis/commits?author=geekazodium) | ✅ |
| @Kepler-17c | [1 commit](https://github.com/Wynntils/Artemis/commits?author=Kepler-17c) | ✅ |

**Comment by magicus** [comment]
I agree

**Comment by xCykrix** [comment]
I agree

**Comment by mbnkwm** [comment]
I agree

**Comment by Rafii2198** [comment]
I agree

**Comment by ShadowCat117** [comment]
I agree

**Comment by iwei20** [comment]
I agree

**Comment by ryanzhoudev** [comment]
I agree

**Comment by cmkohnen** [comment]
I agree

**Comment by bricefrisco** [comment]
I agree

**Comment by DevChromium** [comment]
I agree

**Comment by kristofbolyai** [comment]
I agree

**Comment by TomaSajt** [comment]
I agree

**Comment by derivated** [comment]
I agree

**Comment by McPlayHD** [comment]
I agree

**Comment by atiedebee** [comment]
I agree

**Comment by coehlrich** [comment]
I agree

**Comment by kristofbolyai** [comment]
@KaspianDev Do you reject this..? Why the thumbs down?

**Comment by P0keDev** [comment]
I agree

**Comment by ShadowShift** [comment]
I agree

**Comment by san7890** [comment]
I agree

I always thought the AGPL was overkill, nice to see that you guys are planning on fixing it.

**Comment by Jimmy-sheep** [comment]
I agree, thank you.

**Comment by ENORMOUZ** [comment]
I agree

**Comment by CringeAura** [comment]
I agree

**Comment by re-vc** [comment]
I agree

**Comment by ekgame** [comment]
I agree

**Comment by sofiedotcafe** [comment]
I agree

**Comment by Mcrtin** [comment]
I agree

**Comment by DevScyu** [comment]
I agree

**Comment by KaspianDev** [comment]
> @KaspianDev Do you reject this..? Why the thumbs down?

I prefered the GPL change but I'll think about this one and confirm in a few hours.

**Comment by Zatzou** [comment]
I agree

**Comment by KaspianDev** [comment]
I agree

**Comment by byBackfish** [comment]
I agree

**Comment by Incompleteusern** [comment]
I agree

**Comment by HalcyonSuoh** [comment]
I agree

**Comment by DalwynWasTaken** [comment]
I agree

**Comment by TUsama** [comment]
I agree

**Comment by joshieman06** [comment]
I agree

**Comment by HighCrit** [comment]
I agree

**Comment by YouSoPugly** [comment]
I agree

**Comment by obj-obj** [comment]
I agree

**Comment by Kepler-17c** [comment]
I agree

**Comment by geekazodium** [comment]
I agree

**Comment by magicus** [comment]
Seeing that all contributors have accepted the license change, we will now go ahead with it.

**Comment by magicus** [comment]
This was implemented in #1988. Thank you all for your support. I will now close this bug, but it will continue to serve as a record of contributor acceptance to the license change.
